From bd7817e3f0f5ea7b9188788f124471fc0223c251 Mon Sep 17 00:00:00 2001
From: John Brooks <john.brooks@dereferenced.net>
Date: Mon, 7 Nov 2016 21:29:58 -0700
Subject: [PATCH] Forcefully disable clock_gettime on macOS due to a SDK bug

clock_gettime were added in macOS 10.12, but the __OSX_AVAILABLE_STARTING
macro was left off of the declaration in the SDK header, so the
-mmacosx-version-min flag is ignored. This results in a libevent that links
to symbols only available on 10.12 and later.

A cleaner, possibly upstreamable fix would be to check whether
(MAC_OS_X_VERSION_MIN_REQUIRED < MAC_OS_X_VERSION_10_12) and skip this
function test.
---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index d42edd8..c616d8b 100644
--- a/configure.ac
+++ b/configure.ac
@@ -289,7 +289,7 @@ AC_C_INLINE
 AC_HEADER_TIME
 
 dnl Checks for library functions.
-AC_CHECK_FUNCS([gettimeofday vasprintf fcntl clock_gettime strtok_r strsep])
+AC_CHECK_FUNCS([gettimeofday vasprintf fcntl strtok_r strsep])
 AC_CHECK_FUNCS([getnameinfo strlcpy inet_ntop inet_pton signal sigaction strtoll inet_aton pipe eventfd sendfile mmap splice arc4random arc4random_buf issetugid geteuid getegid getprotobynumber setenv unsetenv putenv sysctl])
 AC_CHECK_FUNCS([umask])
 
-- 
2.10.0

